File: Exercism/fsharp/difference-of-squares/DifferenceOfSquares.fs
module DifferenceOfSquares

let square number = pown number 2

let squareOfSum number = seq { 1 .. number } |> Seq.sum |> square

let sumOfSquares number = seq { 1 .. number } |> Seq.sumBy square

let differenceOfSquares number =
    squareOfSum number - sumOfSquares number

File: Exercism/csharp/difference-of-squares/DifferenceOfSquares.cs
using System.Collections.Generic;
using System.Linq;

public static class DifferenceOfSquares
{
    public static int CalculateSquareOfSum(int max) => max
      .GetRange()
      .Sum()
      .Square();

    public static int CalculateSumOfSquares(int max) => max
      .GetRange()
      .Sum(Square);

    public static int CalculateDifferenceOfSquares(int max) => CalculateSquareOfSum(max) - CalculateSumOfSquares(max);

    public static int Square(this int n) => n * n;

    public static IEnumerable<int> GetRange(this int n) => Enumerable.Range(1, n);
}